packages/webapp/webapp_server.js

```diff
@@ -860,24 +869,46 @@ function runWebAppServer() {
     WebAppInternals.generateBoilerplate();
 
     // only start listening after all the startup code has run.
-    var localPort = WebAppInternals.parsePort(process.env.PORT) || 0;
+    var localPort = process.env.PORT || 0;
     var host = process.env.BIND_IP;
     var localIp = host || '0.0.0.0';
-    httpServer.listen(localPort, localIp, Meteor.bindEnvironment(function() {
-      if (process.env.METEOR_PRINT_ON_LISTEN) {
-        console.log("LISTENING"); // must match run-app.js
-      }
 
-      var callbacks = onListeningCallbacks;
-      onListeningCallbacks = null;
-      _.each(callbacks, function (x) { x(); });
+    var startHttpServer = function(listenOptions) {
+      httpServer.listen(listenOptions, Meteor.bindEnvironment(function() {
+        if (process.env.METEOR_PRINT_ON_LISTEN)
```

Comment:
Sure thing - I didn't really review that part of the code; I'll update that entire function a bit.